File: CPP/program/Car.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"


// Deklarasi namespace
using namespace std;

class Car : public Vehicle{
private:
    string model;
    int jumlahKursi;
    int jumlahPintu;
    
public:
    // Constructor
    Car()
    {
    }
    
    // Constructor dengan parameter lengkap
    Car(string model, int jumlahKursi, int jumlahPintu, string platNomor, string nama, string merk, int tahunProduksi, string warna, string jenisTransmisi) : Vehicle(platNomor, nama, merk, tahunProduksi, warna, jenisTransmisi){
        this->model = model;
        this->jumlahKursi = jumlahKursi;
        this->jumlahPintu = jumlahPintu;
    }


    // Getter untuk mengambil nilai atribut model
    string getModel() {
        return model;
    }

    // Setter untuk mengatur nilai atribut model
    void setModel(string model) {
        this->model = model;
    }

    // Getter untuk mengambil nilai atribut jumlahPintu
    int getJumlahPintu() {
        return jumlahPintu;
    }

    // Setter untuk mengatur nilai atribut jumlahPintu
    void setJumlahPintu(int jumlahPintu) {
        this->jumlahPintu = jumlahPintu;
    }

    // Getter untuk mengambil nilai atribut jumlahKursi
    int getJumlahKursi() {
        return jumlahKursi;
    }

    // Setter untuk mengatur nilai atribut jumlahKursi
    void setJumlahKursi(int jumlahKursi) {
        this->jumlahKursi = jumlahKursi;
    }
    
    // Destructor
    ~Car()
    {
    }
};

File: Car.py
# Kelas Car
from Vehicle import Vehicle

class Car(Vehicle):
    __model=""
    __jumlahKursi=0
    __jumlahPintu=0
    
    # Constructor
    def __init__(self, model="", jumlahKursi=0, jumlahPintu=0, platNomor="", nama="", merk="", tahunProduksi=0, warna="", jenisTransmisi="") : 
        super().__init__(platNomor, nama, merk, tahunProduksi, warna, jenisTransmisi)
        self.__model = model
        self.__jumlahKursi = jumlahKursi
        self.__jumlahPintu = jumlahPintu

    # Getter untuk mengambil nilai atribut model
    def getModel(self) :
        return self.__model

    # Setter untuk mengatur nilai atribut model
    def setModel(self, model) :
        self.__model = model

    # Getter untuk mengambil nilai atribut jumlahPintu
    def getJumlahPintu(self) :
        return self.__jumlahPintu

    # Setter untuk mengatur nilai atribut jumlahPintu
    def setJumlahPintu(self, jumlahPintu) :
        self.__jumlahPintu = jumlahPintu

    # Getter untuk mengambil nilai atribut jumlahKursi
    def getJumlahKursi(self) :
        return self.__jumlahKursi

    # Setter untuk mengatur nilai atribut jumlahKursi
    def setJumlahKursi(self, jumlahKursi) :
        self.__jumlahKursi = jumlahKursi